### PR TITLE
[tfgan] Respect use_loss_summaries in GANEstimator

### PR DESCRIPTION
**Pull request by lgeiger**
Since the refactor done in 47dea684efa41981e10299c2737317c504ce41af the `use_loss_summaries` argument of GANEstimator isn't respected anymore.
This PR restores the original behavior and passes `use_loss_summaries` down to the loss functions.

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by caisq** [comment]
Can you review this please, @joel-shor ?

File: tensorflow/contrib/gan/python/estimator/python/gan_estimator_impl.py [hunk 1]
```diff
@@ -214,12 +214,18 @@ def _get_gan_model(
...
     mode, gan_model, generator_loss_fn, discriminator_loss_fn,
     get_eval_metric_ops_fn, generator_optimizer, discriminator_optimizer,
-    get_hooks_fn=None):
+    get_hooks_fn=None, use_loss_summaries=True):
   """Get the EstimatorSpec for the current mode."""
   if mode == model_fn_lib.ModeKeys.PREDICT:
     estimator_spec = model_fn_lib.EstimatorSpec(
         mode=mode, predictions=gan_model.generated_data)
   else:
+    if 'add_summaries' in inspect.getargspec(generator_loss_fn).args:
```
Comment (joel-shor) [inline tensorflow/contrib/gan/python/estimator/python/gan_estimator_impl.py]: The previous code didn't have this check. Furthermore, the losses test checks that this arg is present: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/gan/python/losses/python/losses_impl_test.py#L158

How would you feel about removing this argument?

**Review by joel-shor — CHANGES_REQUESTED** [review]
Please see the comment I made about checking for `add_summaries` using `inspect`.

File: tensorflow/contrib/gan/python/estimator/python/gan_estimator_impl.py (hunk 1 above)
Comment (lgeiger) [reply tensorflow/contrib/gan/python/estimator/python/gan_estimator_impl.py]: I'm happy to remove this check.
I wasn't sure since the estimator tests failed otherwise due to the missing argument in the dummy loss used there.

Comment (joel-shor) [reply tensorflow/contrib/gan/python/estimator/python/gan_estimator_impl.py]: Ah good point. Would you mind adding that argument to the dummy loss function? Sorry for that, and great catch!

**Comment by lgeiger** [comment]
@joel-shor Sorry for the delay, I removed the check for `add_summaries` using `inspect`.

**Review by joel-shor — APPROVED** [review]

**Comment by joel-shor** [comment]
Great catch, and good fix! Thanks, @lgeiger

**Comment by joel-shor** [comment]
@lgeiger would you mind investigating the test failures?

**Comment by lgeiger** [comment]
This looked like random time-outs, not related to this PR. Thanks for merging 👍
